atom/browser/api/atom_api_web_contents.cc

```diff
@@ -789,9 +790,14 @@ void WebContents::AddNewContents(content::WebContents* source,
     if (disposition != WindowOpenDisposition::NEW_WINDOW &&
         disposition != WindowOpenDisposition::NEW_POPUP) {
       if (!browser) {
-        // TODO(bridiver) - this should be the source owner_window
-        browser = owner_window()->browser();
-        tab_helper->SetWindowId(browser->session_id().id());
+        auto relay = atom::NativeWindowRelay::FromWebContents(source);
+        if (relay) {
+          owner_window = relay->window.get();
+          browser = owner_window->browser();
+          tab_helper->SetWindowId(browser->session_id().id());
+        } else {
+          return;
```

Comment:
what is the case of null relay?
And that means window id will not be set properly in that case

Reply:
This happens when the `owner_window` for the `source` web-contents is not set. I am not sure if there is a way to wait for `source` owner to be set before `AddNewContents` for the new window.

Reply:
the `return` here will cause the webcontents to be leaked and will also prevent b-l from finding out that it ever existed

Reply:
We can initiate a `destroy` here for `new_web_contents`?

Reply:
why would we just arbitrarily decide to destroy the new webcontents here?

Reply:
Also, `else-return` is an anti-pattern. Early return should not have else after it and should test the exceptional condition (assuming null relay were correct, `if (!relay) return;` -- but this is all just an aside on control flow style).